tutorials/examples/cov3d_experiment.dontrun

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env  python
+"""
+This script illustrates Cov3D analysis using experimental dataset
+"""
+import numpy as np
+
+from aspire.basis import FBBasis3D
```

Comment:
FFBBasis3D?

Reply:
This is from Cov3D simulation example. It looks that we did not use FFBBasis3D previously for Cov3D. Got some error probably due to data type inconsistency as below:
 
File "/tigress/junchaox/software/ASPIRE-Python_git/tutorials/examples/cov3d_experiment.dontrun", line 54, in <module>
    mean_est = mean_estimator.estimate()
  File "/tigress/junchaox/software/ASPIRE-Python_git/src/aspire/reconstruction/estimator.py", line 69, in estimate
    est_coeff = self.conj_grad(b_coeff, tol=tol)
  File "/tigress/junchaox/software/ASPIRE-Python_git/src/aspire/reconstruction/estimator.py", line 124, in conj_grad
    operator, b_coeff, M=M, callback=cb, tol=tol, atol=0
  File "<decorator-gen-3>", line 2, in cg
  File "/home/junchaox/.conda/envs/aspire/lib/python3.6/site-packages/scipy/_lib/_threadsafety.py", line 46, in caller
    return func(*a, **kw)
  File "/home/junchaox/.conda/envs/aspire/lib/python3.6/site-packages/scipy/sparse/linalg/isolve/iterative.py", line 315, in cg
    work[slice2] += sclr1*matvec(work[slice1])
TypeError: Cannot cast ufunc add output from dtype('complex64') to dtype('float32') with casting rule 'same_kind'


This might be related to issue

Reply:
As I've said before, the non-fast versions of the FB basis objects exist purely as reference code. We shouldn't use them unless there's a very compelling reason. Is the data type issue due to the tutorial script or some bug in the FFBBasis3D code?

Reply:
I took a look during the passed two days. Both `FFBBasis3D` and `FBBasis3D` generate `float32` type, the `conj_grad` function in the `Estimator` class also does not need `complex64`.  During the interations of CG method, however,  `sclr1matvec(work[slice1])` become `complex64` and needs to be assigned to `work[slice2]` which is float32. Will spend more time on it.

Reply:
I am looking at this now.  The type issue is minor and I think the best way to fix it is to take the real component of the convolution result.  FFB_3d's evaluate_t simply does a dtype pass though. We should not pass it complex numbers.

However, I still encounter a deeper issue with matrix (ndim) sizes trying to run this code. We are passing some things around that are not consistent with the documentation for the methods, so I am debugging that now. It doesn't happen to the unit test fwiw.

Reply:
Took a look again this  morning after adding an explicit real converting, `vol = np.real(kernel.convolve_volume(vol))`. Estimating mean value is fine now. But It looks that the shapes got mismatched when estimating 3D covariance.
   
> covar_est = covar_estimator.estimate(mean_est, noise_variance)
  File "/tigress/junchaox/software/ASPIRE-Python_git/src/aspire/covariance/covar.py", line 93, in estimate
    b_coeff = self.src_backward(mean_vol, noise_variance)
  File "/tigress/junchaox/software/ASPIRE-Python_git/src/aspire/covariance/covar.py", line 188, in src_backward
    covar_b_coeff = self.basis.mat_evaluate_t(covar_b)
  File "/tigress/junchaox/software/ASPIRE-Python_git/src/aspire/basis/basis.py", line 160, in mat_evaluate_t
    return mdim_mat_fun_conj(X, len(self.sz), 1, self.evaluate_t)
  File "/tigress/junchaox/software/ASPIRE-Python_git/src/aspire/utils/matrix.py", line 146, in mdim_mat_fun_conj
    X = f(X)
  File "/tigress/junchaox/software/ASPIRE-Python_git/src/aspire/basis/ffb_3d.py", line 391, in evaluate_t
    v = v.reshape((*sz_roll, self.count))
ValueError: cannot reshape array of size 50176 into shape (8,8,8,8,98)

Reply:
Yes. that is what I said, thanks. I was able to work around for FFB_3d, but then it broke FB_3d and I haven't had time to return to it. ( I thought I would today, but that is not going to happen).

Reply:
The area I was working on was `mdim_mat_fun_conj`, which is pretty questionable at this point.

Reply:
Ok. Since this isn't a simple fix, I suggest we revert to `FBBasis3D` for this example and make an issue to change it to `FFBBasis3D` once these issues are resolved.

Reply:
Create a new issue, #373. The current version is still using the old `FBBasis3D`. It is ready to merge if this is the only point.